modules/core/src/main/java/org/apache/ignite/internal/processors/security/IgniteSecurityProcessor.java

```diff
@@ -197,21 +201,39 @@ public IgniteSecurityProcessor(GridKernalContext ctx, GridSecurityProcessor secP
      * Updates the package access property to specify the internal Ignite package.
      */
     private void updatePackageAccessProperty() {
-        String packAccess = Security.getProperty("package.access");
-
-        if (!F.isEmpty(packAccess)) {
-            if (!packAccess.contains(IGNITE_INTERNAL_PACKAGE))
-                Security.setProperty("package.access", packAccess + ',' + IGNITE_INTERNAL_PACKAGE);
+        synchronized (SANDBOXED_NODES_COUNTER) {
+            if (SANDBOXED_NODES_COUNTER.incrementAndGet() == 1) {
```

Comment:
Since we compare with 0 in `clearPackageAccessProperty` method `getAndIncrement() == 0` looks better.

Reply:
ok